dsgrid/registry/project_registry_manager.py

```diff
@@ -267,6 +280,114 @@ def _register_project_and_dimensions(
         self._register(config, submitter, log_message, force)
         context.add_id(RegistryType.PROJECT, config.model.project_id, self)
 
+    def _register_dimensions_from_models(
+        self,
+        src_dir,
+        dimensions: List,
+        dimension_references,
+        context,
+        submitter,
+        log_message,
+        force,
+    ):
+        dim_model = DimensionsConfigModel(dimensions=dimensions)
+        dims_config = DimensionsConfig.load_from_model(dim_model, src_dir)
+        dimension_ids = self._dimension_mgr.register_from_config(
+            dims_config, submitter, log_message, force=force, context=context
+        )
+        # Order of the next two is required for Pydantic validation.
+        dimension_references += self._dimension_mgr.make_dimension_references(dimension_ids)
+        dimensions.clear()
+
+    def _register_all_in_one_dimensions(
+        self,
+        src_dir,
+        model,
+        context,
+        submitter,
+        log_message,
+        force,
+    ):
+        new_dimensions = []
+        new_mappings = {}
+        dim_type_to_ref = {x.dimension_type: x for x in model.dimensions.base_dimension_references}
+        for dimension_type in model.dimensions.all_in_one_supplemental_dimensions:
+            if dimension_type == DimensionType.TIME:
+                raise DSGInvalidParameter("Cannot have time in all_in_one_supplemental_dimensions")
+            if dimension_type in new_mappings:
+                raise DSGInvalidParameter(
+                    f"{dimension_type} is listed twice in all_in_one_supplemental_dimensions"
+                )
+
+            dim_ref = dim_type_to_ref[dimension_type]
+            dim_config = self._dimension_mgr.get_by_id(dim_ref.dimension_id)
+            supp_dim_dir = src_dir / "dimensions" / "supplemental"
+            if not supp_dim_dir.exists():
+                raise DSGInvalidParameter(
+                    f"supplemental dimensions directory {supp_dim_dir} does not exist"
+                )
```

Comment:
Is this checking for a file path in the registry? And are we confident that src_dir is where all the project information is? If so, should we just make this directory if it doesn't exist?

Reply:
This is not in the registry. This relates to the comment I just posted about new directory requirements.